src/Microsoft.TestPlatform.Common/Utilities/Debug.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics;
+
+using SystemDebug = System.Diagnostics.Debug;
+
+namespace Microsoft.TestPlatform;
+
+[SuppressMessage("ApiDesign", "RS0030:Do not used banned APIs", Justification = "Replacement API to allow nullable hints for compiler")]
+internal static class Debug
```

Comment:
I am always forcing this one but FYI in recent versions the system `Debug.Assert` methods already support it.